testing/e2e/helpers/object.go

```diff
@@ -0,0 +1,52 @@
+//go:build integration
+// +build integration
+
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cartov1alpha1 "github.com/vmware-tanzu/apps-cli-plugin/pkg/apis/cartographer/v1alpha1"
+)
+
+// TODO Improve the comparing of workloads with the autogenerated fields/dates
+func CompareObject(o1, o2 client.Object) bool {
```

Comment:
so while comparing workload objects we care about 
- in metadata section (everything other than timestamp)
- all of spec
- ignore status
- We can use `cmp.Equal` for object metadata and pass ignoretimestamp as the option. 
-